src/roiextractors/imagingextractor.py

```diff
@@ -67,8 +67,10 @@ def get_video(
         pass
 
     def get_frames(self, frame_idxs: ArrayType, channel: Optional[int] = 0) -> np.ndarray:
-
-        return self.get_video()[frame_idxs, ..., channel]  # To-do simplify for contigious frame_idxs case
+        if np.all(np.diff(frame_idxs) == 0):
+            return self.get_video(start_frame=frame_idxs[0], end_frame=frame_idxs[-1])
+        relative_indices = np.array(frame_idxs) - frame_idxs[0]
+        return self.get_video(start_frame=frame_idxs[0], end_frame=frame_idxs[-1])[relative_indices, ..., channel]
```

Comment:
However, the original `self.get_video()` call almost crashed my computer by attempting to load an entire video into memory just to do the fancy indexing on it - all in order to determine the `dtype` from the first frame. If the return type from `get_video` is a memmap this obviously isn't a problem, but it may not always be guaranteed to be.

This correction only loads the amount of video needed to perform the fancy indexing.